junit/src/main/java/org/kaazing/k3po/junit/rules/Latch.java

```diff
@@ -28,6 +29,8 @@
     private final CountDownLatch startable;
     private final CountDownLatch finished;
     private volatile Thread testThread;
+    
+    private final AtomicBoolean exceptionOccurred = new AtomicBoolean(false); 
```

Comment:
I will clean up `setInterruptOnException` to set only the handler.

I also realized that `awaitBarrier` and `notifyBarrier` are only called from the main test thread. Because of this, the exception should not be handled using the Latch (`Latch.notifyException()` is used to communicate an exception that occurred in the communication thread to the main thread, not the other way around), but it should just be thrown. It will be caught in `evaluate()` and handled there properly.

As for the catch blocks in the statement, I'm still working on them, to try to unify the code. First I need to make sure we cover all situations.

I hope to commit a new version by EOD.